Supertext.Base.Net/Mail/MailService.cs

```diff
@@ -44,6 +47,39 @@ await SendInternalAsync(mail,
             }
         }
 
+        public async Task SendUsingTemplateAsync(EmailInfoTemplates mailInfo, CancellationToken ct = default)
+        {
+            try
+            {
+                var options = new SendGridClientOptions
+                {
+                    ApiKey = _mailServiceConfig.SendGridPassword
+                };
+                var client = new SendGridClient(options);
+                var message = new SendGridMessage
+                              {
+                                  TemplateId = mailInfo.TemplateId,
+                                  From = new EmailAddress(mailInfo.From.Email, mailInfo.From.Name),
+                                  Subject = mailInfo.Subject
+                              };
+                message.SetTemplateData(mailInfo.DynamicTemplateDataAsJson);
+                message.AddTos(mailInfo.Recipients.Select(r => new EmailAddress(r.Email, r.Name)).ToList());
```

Comment:
Just an idea: How about a static extension method which converts `PersonInfo` to `EmailAddress`? Could be used on lines 62 and 66.

Reply:
refined